hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMContext.java

```diff
@@ -96,6 +96,38 @@ public void updateLeaderAndTerm(boolean leader, long newTerm) {
     }
   }
 
+  /**
+   * Update isLeader flag.
+   * @param leader
+   */
+  public void updateLeader(boolean leader) {
+    lock.writeLock().lock();
+    try {
```

Comment:
This looks a bit confusing. It should be ideally leaderReady. Let's maintain notion of leader and leaderReady separately.

Reply:
Why do we need 2 leader and leader ready. Actually we need leader ready every where.
Do you see any use-case for both?

Reply:
Updated to fix this confusion and also added 2 APIs.